tableauserverclient/filesys_helpers.py

```diff
@@ -20,3 +20,15 @@ def make_download_path(filepath, filename):
         download_path = filepath + os.path.splitext(filename)[1]
 
     return download_path
+
+
+def file_is_compressed(file):
```

Comment:
This is the main hang up I have, I have a feeling that peeking into file contents just to get the extension feels wrong.

But I know that the publish requires you specify the file type and I don’t have an alternative in mind to suggest.

Curious, did you have any other ideas you rejected for this issue?

Reply:
I would also prefer not to specify file type in the API call but can't really get around that. As the file object is not saved we neither have any metadata to look into. Only alternative that I could think of was to pass file type as an optional parameter for the publish method and require it when dealing with file objects. The reason I didn't go for this was that I preferred to restrict the changes to function internals as opposed to changing its interface as well. Now thinking about it, adding one optional parameter probably wouldn't  hurt if you would prefer it over sniffing the file contents.

Reply:
Alright, fair enough.

Let's keep the peeking approach